### PR TITLE
added new operations in string-utils  for converting strings in url e…

### PR DESCRIPTION
**Pull request by klag**
…ncoded strings

### CODE REVIEW

File: javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java [hunk 1]
```diff
@@ -325,4 +326,31 @@ public String toUpperCase( String str ) {
 	public Boolean contains( Value request ) {
 		return request.strValue().contains( request.getFirstChild( "substring" ).strValue() );
 	}
+
+	public String urlEncode( Value request ) throws FaultException {
+		String characterEncoding = "UTF-8";
```
Comment (fmontesi) [inline javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java:331]: You can rewrite 331-334 with:

```
String charset = request.firstChildOrDefault( "charset", Value::strValue, "UTF-8" );
```

File: javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java [hunk 2]
```diff
@@ -325,4 +326,31 @@ public String toUpperCase( String str ) {
...
+		if( request.getFirstChild( "characterEncoding" ).isDefined() ) {
+			characterEncoding = request.getFirstChild( "characterEncoding" ).strValue();
+		}
+		try {
+			String response = URLEncoder.encode( request.strValue(), characterEncoding );
```
Comment (fmontesi) [inline javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java:336]: Just return it:

```
return URLEncoder.encode( request.strValue(), characterEncoding );
```

File: javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java [hunk 3]
```diff
@@ -325,4 +326,31 @@ public String toUpperCase( String str ) {
...
+			return response;
+		} catch( UnsupportedEncodingException e ) {
+			throw new FaultException( "UnsupportedEncodingException" );
+		}
+
+	}
+
+	public String urlDecode( Value request ) throws FaultException {
+		String characterEncoding = "UTF-8";
```
Comment (fmontesi) [inline javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java:345]: See comments for previous methods.

**Review by fmontesi — CHANGES_REQUESTED** [review]

File: javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java (hunk 2 above)
Comment (klag) [reply javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java:336]: What about the fault exception

File: javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java [hunk 4]
```diff
@@ -325,4 +326,29 @@ public String toUpperCase( String str ) {
 	public Boolean contains( Value request ) {
 		return request.strValue().contains( request.getFirstChild( "substring" ).strValue() );
 	}
+
+	public String urlEncode( Value request ) throws FaultException {
+		String charset = request.firstChildOrDefault( "charset", Value::strValue, "UTF-8" );
+		if( request.getFirstChild( "characterEncoding" ).isDefined() ) {
```
Comment (fmontesi) [inline javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java:332]: remove.

File: include/string_utils.iol [hunk 5]
```diff
@@ -82,6 +82,12 @@ type ContainsRequest:string {
 	.substring:string
 }
 
+type UrlEncodeRequest: string {
+	characterEncoding?: string
```
Comment (fmontesi) [inline include/string_utils.iol:86]: why not charset?

File: javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java [hunk 6]
```diff
@@ -325,4 +326,29 @@ public String toUpperCase( String str ) {
...
+			return URLEncoder.encode( request.strValue(), charset );
+		} catch( UnsupportedEncodingException e ) {
+			throw new FaultException( "UnsupportedEncodingException" );
+		}
+
+	}
+
+	public String urlDecode( Value request ) throws FaultException {
+		String charset = request.firstChildOrDefault( "charset", Value::strValue, "UTF-8" );
+		if( request.getFirstChild( "characterEncoding" ).isDefined() ) {
```
Comment (fmontesi) [inline javaServices/coreJavaServices/src/main/java/joliex/util/StringUtils.java:345]: remove.

File: packages/string-utils.ol [hunk 7]
```diff
@@ -82,6 +82,12 @@ type ContainsRequest:string {
 	.substring:string
 }
 
+type UrlEncodeRequest: string {
+	characterEncoding?: string
```
Comment (fmontesi) [inline packages/string-utils.ol:86]: why not charset?

**Review by fmontesi — CHANGES_REQUESTED** [review]

**Pull request merged**